### PR TITLE
Add verification of issuer signing key with integration test

### LINKED ISSUE #2323: [Feature Request] Use the new AzureAD key issuer validator in AddMicrosoftIdentityWebApi
**Is your feature request related to a problem? Please describe.**
IdentityModel now exposes a new AAD specific implementation for a delegate validating keys issuer. AddMicrosoftIdentityWebApi could benefit from it.
For details, see https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2134

**Describe the solution you'd like**
Add the validation of the keys issuer, by calling `mergedOptions.TokenValidationParameters.EnableAadSigningKeyIssuerValidation()` for instance [here](https://github.com/AzureAD/microsoft-identity-web/blob/bd99dcd83e2fac586064a69b377fbe550ab5484f/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs#L218)

### PR DESCRIPTION
**Pull request by jmprieur**
# Add verification of issuer signing key with integration test

- Fixes AddMicrosoftWebApi() to pass-in the configuration from the JwtBearer options to the token validation parameters.
- Adds a SimulateOidc project under IntegrationTests to generate invalid sign-in key issuer for the negative test (the authority
  is then exposed to https://localhost:1234/v2.0
- Updates an integration test to use the test authority
- Factorize the code that starts an external app to Microsoft.Identity.Web.Test.Commons (ExternalApp.Start)

Fixes #2323 

> Note that the logs of the IntegrationTestService show the invalid signing key issuer failing the request.
> fail: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
>       IDX40005: Token issuer: 'https://login.microsoftonline.com/f645ad92-e38d-4d1a-b510-d1b09a74a8ca/v2.0', does not match the signing key issuer: 'invalidIssuer'.

### CODE REVIEW

File: src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs [hunk 1]
```diff
@@ -176,6 +177,7 @@ public static class MicrosoftIdentityWebApiAuthenticationBuilderExtensions
                     msIdOptionsMonitor.Get(jwtBearerScheme); // needed for firing the PostConfigure.
                     MergedOptions mergedOptions = mergedOptionsMonitor.Get(jwtBearerScheme);
 
+
                     if (mergedOptions.Authority != null)
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs:181]: nit: remove line

File: tests/DevApps/WebAppCallsWebApiCallsGraph/Client/appsettings.json [hunk 2]
```diff
@@ -10,7 +10,6 @@
     "EnableCacheSynchronization": false,
 
     // To call an API
-    "ClientSecret": "secret"
```
Comment (jennyf19) [inline tests/DevApps/WebAppCallsWebApiCallsGraph/Client/appsettings.json:13]: > "ClientSecret": "secret" [](http://example.com/codeflow?start=3&length=25)

can you add this back? I like to use it.

File: tests/DevApps/WebAppCallsWebApiCallsGraph/TodoListService/appsettings.json [hunk 3]
```diff
@@ -1,8 +1,15 @@
 {
     "AzureAd": {
-        "Instance": "https://login.microsoftonline.com/",
-        "Domain": "msidentitysamplestesting.onmicrosoft.com",
-        "TenantId": "7f58f645-c190-4ce5-9de4-e2b7acd2a6ab",
+        "Authority": "https://login.microsoftonline.com/7f58f645-c190-4ce5-9de4-e2b7acd2a6ab/",
```
Comment (jennyf19) [inline tests/DevApps/WebAppCallsWebApiCallsGraph/TodoListService/appsettings.json:3]: >Authority" [](http://example.com/codeflow?start=9&length=10)

can use Instance instead right?

File: tests/DevApps/WebAppCallsWebApiCallsGraph/TodoListService/Startup.cs [hunk 4]
```diff
@@ -65,6 +65,14 @@ public void ConfigureServices(IServiceCollection services)
                      .AddInMemoryTokenCaches();
 #endif
 
+            services.Configure<JwtBearerOptions>(JwtBearerDefaults.AuthenticationScheme,
+     options =>
+     {
+         options.RequireHttpsMetadata = false;
+     }
+
```
Comment (jennyf19) [inline tests/DevApps/WebAppCallsWebApiCallsGraph/TodoListService/Startup.cs:73]: remove line and check spacing

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: tests/DevApps/WebAppCallsWebApiCallsGraph/TodoListService/Startup.cs (hunk 4 above)
Comment (jmprieur) [reply tests/DevApps/WebAppCallsWebApiCallsGraph/TodoListService/Startup.cs:73]: Removed as it's not needed. The SimulateOidc runs on https:

**Pull request merged**